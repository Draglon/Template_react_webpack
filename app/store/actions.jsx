export const POPUP_TOGGLE = 'POPUP_TOGGLE';
export const MENU_TOGGLE = 'MENU_TOGGLE';

export const popupToggle = () => {
    return {
        type: POPUP_TOGGLE
    };
};

export const menuToggle = () => {
    return {
        type: MENU_TOGGLE
    };
};