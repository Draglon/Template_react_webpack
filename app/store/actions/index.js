export * from './ui';
export * from './login';
