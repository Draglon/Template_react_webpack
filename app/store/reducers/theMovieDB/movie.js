import { MOVIE_REQUEST, MOVIE_SUCCESS, MOVIE_FAILURE } from '../../constants/movie.constants';

const movieInitialState = {
  isPending: false,
  data: null,
  error: null,
};

// Movie
const movieReducer = (state = movieInitialState, action) => {
  switch (action.type) {
    case MOVIE_REQUEST:
      return {
        ...state,
        isPending: true,
        error: null,
      };
    case MOVIE_SUCCESS:
      return {
        ...state,
        isPending: false,
        data: action.payload,
        error: null,
      };
    case MOVIE_FAILURE:
      return {
        ...state,
        isPending: false,
        error: action.payload,
      };
    default:
      return state;
  }
};

export default movieReducer;
