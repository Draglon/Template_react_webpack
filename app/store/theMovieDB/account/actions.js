import * as t from './actionTypes';

export const profileRequest = () => ({ type: t.PROFILE_REQUEST });
export const profileSuccess = payload => ({ type: t.PROFILE_SUCCESS, payload });
export const profileFailure = payload => ({ type: t.PROFILE_FAILURE, payload });
