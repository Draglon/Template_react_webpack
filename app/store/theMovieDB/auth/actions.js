import * as t from './actionTypes';

export const createSessionRequest = payload => ({ type: t.CREATE_SESSION_REQUEST, payload });
export const createSessionSuccess = payload => ({ type: t.CREATE_SESSION_SUCCESS, payload });
export const createSessionFailure = payload => ({ type: t.CREATE_SESSION_FAILURE, payload });

export const deleteSessionRequest = () => ({ type: t.DELETE_SESSION_REQUEST });
export const deleteSessionSuccess = payload => ({ type: t.DELETE_SESSION_SUCCESS, payload });
export const deleteSessionFailure = payload => ({ type: t.DELETE_SESSION_FAILURE, payload });
