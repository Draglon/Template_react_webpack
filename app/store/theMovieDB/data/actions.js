import * as t from './actionTypes';

export const addEntities = entities => ({ type: t.ADD_ENTITIES, entities });
