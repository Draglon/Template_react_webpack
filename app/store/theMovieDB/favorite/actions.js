import * as t from './actionTypes';

export const favoriteRequest = payload => ({ type: t.FAVORITE_REQUEST, payload });
export const favoriteSuccess = payload => ({ type: t.FAVORITE_SUCCESS, payload });
export const favoriteFailure = payload => ({ type: t.FAVORITE_FAILURE, payload });

export const addToFavoriteRequest = payload => ({ type: t.ADD_TO_FAVORITE_REQUEST, payload });
export const addToFavoriteSuccess = payload => ({ type: t.ADD_TO_FAVORITE_SUCCESS, payload });
export const addToFavoriteFailure = payload => ({ type: t.ADD_TO_FAVORITE_FAILURE, payload });
