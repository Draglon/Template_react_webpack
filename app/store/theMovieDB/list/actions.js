import * as t from './actionTypes';

// Create list
export const createListRequest = payload => ({ type: t.CREATE_LIST_REQUEST, payload });
export const createListSuccess = payload => ({ type: t.CREATE_LIST_SUCCESS, payload });
export const createListFailure = payload => ({ type: t.CREATE_LIST_FAILURE, payload });

// Add movie
// export const addMovieRequest = payload => ({ type: t.ADD_MOVIE_REQUEST, payload });
// export const addMovieSuccess = payload => ({ type: t.ADD_MOVIE_SUCCESS, payload });
// export const addMovieFailure = payload => ({ type: t.ADD_MOVIE_FAILURE, payload });
