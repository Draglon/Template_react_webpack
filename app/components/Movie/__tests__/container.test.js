import React from 'react';
import configureStore from 'redux-mock-store';
import { movieRequest } from '../../../store/theMovieDB/movie/actions';
import MovieContainer from '../container';

describe('<MovieContainer />', () => {
  const id = 9999;
  const props = {
    match: {
      params: {
        id,
      },
    },
  };
  const state = {
    reducers: {
      data: {
        movie: {
          [id]: {
            id,
            genres: [1, 2, 3],
            cast: [1, 2, 3],
            crew: [1, 2, 3],
          },
        },
        genres: {
          1: { id: 1, name: 'name' },
          2: { id: 2, name: 'name' },
        },
        cast: {
          1: { id: 1, name: 'name', character: 'character' },
          2: { id: 2, name: 'name', character: 'character' },
        },
        crew: {
          1: { id: 1, name: 'name', job: 'position' },
          2: { id: 2, name: 'name', job: 'position' },
        },
      },
    },
  };

  const store = configureStore()(state);
  store.dispatch = jest.fn();

  const wrapper = shallow(<MovieContainer store={store} {...props} />);
  const container = wrapper.dive();
  const instance = container.instance();

  describe('componentDidMount()', () => {
    it('dispatches the `movieRequest()`', () => {
      instance.componentDidMount();
      expect(store.dispatch).toHaveBeenCalledWith(movieRequest({ movieId: id }));
    });
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
