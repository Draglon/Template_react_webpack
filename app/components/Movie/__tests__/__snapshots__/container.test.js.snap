// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieContainer /> should match snapshot 1`] = `
<MovieComponent
  cast={
    Array [
      Object {
        "character": "character",
        "id": 1,
        "name": "name",
      },
      Object {
        "character": "character",
        "id": 2,
        "name": "name",
      },
      undefined,
    ]
  }
  crew={
    Array [
      Object {
        "id": 1,
        "job": "position",
        "name": "name",
      },
      Object {
        "id": 2,
        "job": "position",
        "name": "name",
      },
      undefined,
    ]
  }
  genres={
    Array [
      Object {
        "id": 1,
        "name": "name",
      },
      Object {
        "id": 2,
        "name": "name",
      },
      undefined,
    ]
  }
  match={
    Object {
      "params": Object {
        "id": 9999,
      },
    }
  }
  movie={
    Object {
      "cast": Array [
        1,
        2,
        3,
      ],
      "crew": Array [
        1,
        2,
        3,
      ],
      "genres": Array [
        1,
        2,
        3,
      ],
      "id": 9999,
    }
  }
  movieRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "payload": Object {
                "movieId": 9999,
              },
              "type": "MOVIE_REQUEST",
            },
          ],
          Array [
            Object {
              "payload": Object {
                "movieId": 9999,
              },
              "type": "MOVIE_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "payload": Object {
                  "movieId": 9999,
                },
                "type": "MOVIE_REQUEST",
              },
            ],
            Array [
              Object {
                "payload": Object {
                  "movieId": 9999,
                },
                "type": "MOVIE_REQUEST",
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
